pallets/automation-time/src/tests.rs

```diff
@@ -2125,34 +2156,63 @@ fn trigger_tasks_limits_missed_slots() {
 fn trigger_tasks_completes_all_tasks() {
 	new_test_ext(START_BLOCK_TIME).execute_with(|| {
 		let message_one: Vec<u8> = vec![2, 4, 5];
+		let owner = AccountId32::new(ALICE);
 		let task_id1 = add_task_to_task_queue(
 			ALICE,
 			vec![40],
 			vec![SCHEDULED_TIME],
 			Action::Notify { message: message_one.clone() },
+			vec![],
 		);
 		let message_two: Vec<u8> = vec![2, 4];
 		let task_id2 = add_task_to_task_queue(
 			ALICE,
 			vec![50],
 			vec![SCHEDULED_TIME],
 			Action::Notify { message: message_two.clone() },
+			vec![],
 		);
 		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
 
 		AutomationTime::trigger_tasks(Weight::from_ref_time(120_000));
 
+		let mut condition: BTreeMap<String, String> = BTreeMap::new();
+		condition.insert(String::from("type"), String::from("time"));
+		condition.insert(String::from("timestamp"), format!("{}", LAST_BLOCK_TIME));
+
 		assert_eq!(
 			events(),
 			[
+				RuntimeEvent::AutomationTime(crate::Event::TaskTriggered {
+					who: owner.clone(),
+					task_id: task_id1.clone(),
+					condition: condition.clone(),
+					encoded_call: vec![],
+					abort_errors: vec![],
+				}),
 				RuntimeEvent::AutomationTime(crate::Event::Notify { message: message_one.clone() }),
+				RuntimeEvent::AutomationTime(crate::Event::TaskExecuted {
+					who: owner.clone(),
+					task_id: task_id1.clone(),
+				}),
 				RuntimeEvent::AutomationTime(crate::Event::TaskCompleted {
-					who: AccountId32::new(ALICE),
+					who: owner.clone(),
 					task_id: task_id1.clone(),
 				}),
```

Comment:
`Event::TaskCompleted` is not supposed to fire here. It fires only after all executions of a task is done.

Reply:
Yes, It indeed triggers when the task is completed.

There are 2 tasks in this test, and each task is executed only once. So, after each task is executed, the Event::TaskCompleted will be triggered.

Reply:
okay, please add comments here to indicated that the first task is completed and the subsequent events are for the rest of tasks.